bundles/org.openhab.binding.comfoair/src/main/java/org/openhab/binding/comfoair/internal/ComfoAirSerialConnector.java

```diff
@@ -0,0 +1,652 @@
+        Integer requestCmd = command.getRequestCmd();
+        int retry = 0;
+
+        if (requestCmd != null) {
+            // Switch support for app or ccease control
+            if (requestCmd == 0x9b) {
+                isSuspended = !isSuspended;
+            } else if (requestCmd == 0x9c) {
+                return new int[] { isSuspended ? 0x00 : 0x03 };
+            } else if (isSuspended) {
+                logger.trace("Ignore cmd. Service is currently suspended");
+                return new int[0];
+            }
+
+            do {
+                // If preRequestData param was send (preRequestData is sending for write command)
+                int[] requestData;
+
+                if (preRequestData.length <= 0) {
+                    requestData = command.getRequestData();
+                } else {
+                    requestData = buildRequestData(command, preRequestData);
+
+                    if (requestData.length <= 0) {
+                        logger.debug("Unable to build data for write command: {}",
+                                String.format("%02x", command.getReplyCmd()));
+                        return new int[0];
+                    }
+                }
+
+                // Fake read request for ccease properties
+                if (requestData.length <= 0 && requestCmd == 0x37) {
+                    requestData = new int[] { 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00 };
+                }
+
+                byte[] requestBlock = calculateRequest(requestCmd, requestData);
+                logger.trace("send DATA: {}", dumpData(requestBlock));
+
+                if (!send(requestBlock)) {
+                    return new int[0];
+                }
+
+                byte[] responseBlock = new byte[0];
+
+                try {
+                    InputStream inputStream = this.inputStream;
+                    // 31 is max. response length
+                    byte[] readBuffer = new byte[31];
+
+                    do {
+                        while (inputStream != null && inputStream.available() > 0) {
+
+                            int bytes = inputStream.read(readBuffer);
+
+                            // merge bytes
+                            byte[] mergedBytes = new byte[responseBlock.length + bytes];
+                            System.arraycopy(responseBlock, 0, mergedBytes, 0, responseBlock.length);
+                            System.arraycopy(readBuffer, 0, mergedBytes, responseBlock.length, bytes);
+
+                            responseBlock = mergedBytes;
+                        }
+                        try {
+                            // add wait states around reading the stream, so that
+                            // interrupted transmissions are merged
+                            Thread.sleep(100);
+                        } catch (InterruptedException e) {
+                            // ignore interruption
```

Comment:
Interruptions usually mean that the thread should stop what it is doing and exit. It definitely isn't something that should be ignored in a loop. I would suggest throwing an exception instead, possibly the InterruptedException itself.